src/content/en/updates/posts/2016/04/ump.markdown

```diff
@@ -0,0 +1,156 @@
+
+Here's the relevant code:
+
+{% highlight javascript %}
+var recordedBlobs = [];
+
+mediaRecorder.ondataavailable = function(event) {
+  if (event.data && event.data.size > 0) {
+    recordedBlobs.push(event.data);
+  }
+};
+
+function play() {
+  var superBuffer = new Blob(recordedBlobs, {type: 'video/webm'});
+  recordedVideo.src = window.URL.createObjectURL(superBuffer);
+}
+{% endhighlight %}
+
+For the demo at <a href="https://simpl.info/video/offline" title="Offline video using the File APIs">simpl.info/video/offline</a>, video is stored using the File APIs, then played back using a Blob URL:
+
+<a href="https://simpl.info/video/offline"><img class="screenshot-portrait" src="/web/updates/images/2016/04/ump/screenshot-file.jpg" alt="Screenshot of playback in Chrome on Android of video stored using the File APIs"></a>
+
+{% highlight javascript %}
+function writeToFile(fileEntry, blob) {
+  fileEntry.createWriter(function(fileWriter) {
+    fileWriter.onwriteend = function() {
+      readFromFile(fileEntry.fullPath);
+    };
+    fileWriter.onerror = function(e) {
+      log('Write failed: ' + e.toString());
+    };
+    fileWriter.write(blob);
+  }, handleError);
+}
+
+function readFromFile(fullPath) {
+  window.fileSystem.root.getFile(fullPath, {}, function(fileEntry) {
+    fileEntry.file(function(file) {
+      var reader = new FileReader();
+      reader.onloadend = function() {
+        video.src = URL.createObjectURL(new Blob([this.result]));
+      };
+      reader.readAsArrayBuffer(file);
+    }, handleError);
+  }, handleError);
+}
+{% endhighlight %}
+
+The Unified Media Pipeline has also been [enabled for MSE and EME](https://groups.google.com/a/chromium.org/forum/#!topic/chromium-reviews/Qi4dLcKjcCM).
+
+This is another step towards unifying mobile and desktop Chrome. You don't need to change your code, but building a consistent media experience across desktop and mobile should now be easier, since the media stack is the same across platforms. Debugging with Chrome DevTools? Mobile emulation now uses the 'real' audio and video stack.
+
+If you experience problems as a result of the Unified Media Pipeline, please file issues on the [implementation bug](https://groups.google.com/a/chromium.org/forum/#!topic/chromium-reviews/Qi4dLcKjcCM) or via [new.crbug.com](https://new.crbug.com).
+
+## Demos
+
+* <a href="https://googlechrome.github.io/samples/service-worker/prefetch-video/index.html">Cache video with a service worker</a>
+* <a href="https://simpl.info/video/playbackrate">Media `playbackRate`</a>
+* <a href="https://simpl.info/mediarecorder">MediaRecorder: playback using a Blob URL</a>
+* <a href="https://simpl.info/video/offline">Offline video implemented with the File APIs</a>
```

Comment:
This is the name of a set of APIs: the File APIs. (See [simpl.info/mse](https://simpl.info/mse/) for an example.)